TaskWithInheritanceAndPolymorphism/TaskWithInheritanceAndPolymorphism.cpp

```diff
@@ -0,0 +1,139 @@
+﻿#include <iostream>
+#include <memory>
+#include <vector>
+
+template <typename T>
```

Comment:
Please read and implement https://stackoverflow.com/questions/14294267/class-template-for-numeric-types. I like the static_assert solution.

Reply:
Done